internal/pipeline/pipeline.go

```diff
@@ -29,30 +30,48 @@ var (
 	}
 )
 
-func New(log logger.Handler, informer dynamicinformer.DynamicSharedInformerFactory, broker broker.Handler, plConfigs map[string]internalconfig.PipelineConfigs, stopChan chan struct{}) *pipeline.Pipeline {
+func New(log logger.Handler, informer dynamicinformer.DynamicSharedInformerFactory, broker broker.Handler, plConfigs map[string]internalconfig.PipelineConfigs, stopChan chan struct{}, dynamicKube dynamic.Interface) *pipeline.Pipeline {
+	// TODO: best way to check whether WatchList feature is enabled
+	watchList := true
```

Comment:
I have looked around for ways of determining the enabled features in a Kubernetes cluster, but most knowledge revolves around looking at actual cluster configuration to see what has been enabled and what hasn't been enabled...I would appreciate any pointers to how I can determine programmatically which features have been enabled or not

Reply:
@KiptoonKipkurui I encourage you to review this PR (its in Rust), but potentially a great reference - http://gitea.lntu.edu.cn/youling/kube/commit/3af6b7f1df216380dd9c9bd13a163fe5281d67fa#diff-d4e0d370315574d0ee0602792eb3abfd2777d3c7

I assume that you have already seen this as well - https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/3157-watch-list/README.md

Reply:
Thank you for this resources, I will check them out